parsel/selector.py

```diff
@@ -196,25 +213,27 @@ class Selector(object):
     selectorlist_cls = SelectorList
 
     def __init__(self, text=None, type=None, namespaces=None, root=None,
-                 base_url=None, _expr=None):
+                 base_url=None, _expr=None, json=None):
         self.type = st = _st(type or self._default_type)
         self._parser = _ctgroup[st]['_parser']
         self._csstranslator = _ctgroup[st]['_csstranslator']
         self._tostring_method = _ctgroup[st]['_tostring_method']
+        self.text = text  # save source text is friendly to jpath
```

Comment:
If this is only needed for internal purposes, `self._text` is probably a better choice.

Reply:
Thank you for your valuable suggestions!